app/models/katello/erratum.rb

```diff
@@ -58,7 +58,7 @@ def self.applicable_to_hosts(hosts)
       # which is calculated elsewhere.
 
       self.joins(:content_facets).
-        where("#{Katello::Host::ContentFacet.table_name}.host_id" => hosts.select(:id))
+        where("#{Katello::Host::ContentFacet.table_name}.host_id" => hosts.pluck(:id))
```

Comment:
This has performance implications and based on the error, it appears that the hosts being passed in are not an AR relation, but instead an array.  .pluck() will cause a query to be performed to fetch all the ids, returned rails, just for rails to send the id list back to get the errata.   using .select (on an AR query), will cause it all to be done in the database.  Can you figure out why the hosts list coming in are not an AR query?

Reply:
Seems this is the issue https://github.com/Katello/katello/blob/579a1336044773170cabd5a194194f2eab66aaa1/app/controllers/katello/api/v2/errata_controller.rb#L87

And probably this should fix 
~~~
@host = ::Host::Managed.authorized("view_hosts").where(:id => params[:host_id])
~~~

Reply:
Ah, so yes, i would change the line here: https://github.com/Katello/katello/blob/579a1336044773170cabd5a194194f2eab66aaa1/app/controllers/katello/api/v2/errata_controller.rb#L53
to something like  `::Host::Managed.where(:id => @host.id) ` 

Also a test covering this scenario would be great!